toolchain/check/context.h

```diff
@@ -24,6 +24,40 @@ class Context {
   // A block of code that has not yet been inserted into SemIR.
   class PendingBlock;
 
+  // Description of the target of a conversion.
+  struct ConversionTarget {
```

Comment:
I think at this point, it's important to factor `Convert` and all of this out into a separate class with its own header and source file.

Are there barriers to that? FWIW, if reasonable, I would prefer to do that here before reviewing the full implementation as I think it'll make it easier for me to review as well.

One key thing that I think this will clarify is to introduce a layering between the conversion and the rest of checking, and to understand how that layering works. It may even require some callbacks or something to handle complex aspects of the layering, but that would actually make the code easier to read for me because it would highlight where things can re-enter if it makes sense.

Reply:
Done.